prototype_source/quantization_in_pytorch_2_0_export_tutorial.rst

```diff
@@ -0,0 +1,630 @@
+Taking QNNPackQuantizer as an example, the overall Quantization 2.0 flow could be:
+
+::
+
+    import torch
+    import torch._dynamo as torchdynamo
+    from torch.ao.quantization._quantize_pt2e import convert_pt2e, prepare_pt2e
+    import torch.ao.quantization._pt2e.quantizer.qnnpack_quantizer as qq
+
+    class M(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.linear = torch.nn.Linear(5, 10)
+
+        def forward(self, x):
+            return self.linear(x)
+
+    example_inputs = (torch.randn(1, 5),)
+    model = M().eval()
+
+    # Step 1: Trace the model into an FX graph of flattened ATen operators
+    exported_graph_module, guards = torchdynamo.export(
+        model,
+        *copy.deepcopy(example_inputs),
+        aten_graph=True,
+    )
+
+    # Step 2: Insert observers or fake quantize modules
+    quantizer = qq.QNNPackQuantizer()
+    operator_config = qq.get_symmetric_quantization_config(is_per_channel=True)
+    quantizer.set_global(operator_config)
+    prepared_graph_module = prepare_pt2e_quantizer(exported_graph_module, quantizer)
+
+    # Step 3: Quantize the model
+    convered_graph_module = convert_pt2e(prepared_graph_module)
+
+    # Step 4: Lower Reference Quantized Model into the backend
+
+Inside the Quantizer, we will use the ``QuantizationAnnotation API``
+to convey user's intent for what quantization spec to use and how to
+observe certain tensor values in the prepare step. Now, we will have a step-by-step
+tutorial for how to use the ``QuantizationAnnotation API`` with different types of
+``QuantizationSpec``.
+
+1. Annotate common operator patterns
+--------------------------------------------------------
+
+In order to use the quantized operators, e.g. ``quantized add``,
+backend developers will have intent to quantize (as expressed by
+`QuantizationSpec <https://github.com/pytorch/pytorch/blob/1ca2e993af6fa6934fca35da6970308ce227ddc7/torch/ao/quantization/_pt2e/quantizer/quantizer.py#L38>`__
+) input, output of the operator. Following is an example flow (with ``add``)
+of how this intent is conveyed in the quantization workflow with node annotation API.
+
+-  Step1: Identify the original floating point ``add`` node in the FX graph. There are
+   several ways to identify this node: 1. User may use a pattern matcher (e.g. SubgraphMatcher)
+   to match the operator pattern. 2. User may go through the nodes from start to the end and compare
+   the node's target type.
+-  Step2: Define the ``QuantizationSpec`` for two inputs and one output of the ``add`` node to specify
+   how to quantize input tensors and output tensor which includes parameters of ``observer type``,
+   ``dtype``, ``quant_min``, and ``quant_max`` etc.
+-  Step3: Annotate the inputs and output of the ``add`` node. User will create the ``QuantizationAnnotation``
+   object and add it into ``add`` node's ``meta`` property.
+
+::
+
+    def _annotate_add(
```

Comment:
I feel we do not need to paste this whole code, the tutorial should focus on explaining every line of code instead, we can extract the relevant part and put that in each section and have good explanations for them so that users can take this and implement their own annotation

Reply:
maybe some of the explanations can be in the docstrings, not necessarily here, we don't need to go into a lot of details but mostly focus on explaining how these things are connected here